api_app/analyzers_manager/observable_analyzers/ip2location.py

```diff
@@ -0,0 +1,45 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+
+import requests
+
+from api_app.analyzers_manager import classes
+from api_app.analyzers_manager.exceptions import AnalyzerRunException
+from tests.mock_utils import MockUpResponse, if_mock_connections, patch
+
+
+class Ip2location(classes.ObservableAnalyzer):
+    base_url: str = "https://api.ip2location.io/"
+    _api_key_name: str
+
+    def get_response(self, payload):
+        return requests.get(self.base_url, params=payload)
+
+    def run(self):
+        try:
+            payload = {"ip": self.observable_name}
+
+            # Add key to payload if the API version is "keyed"
+            if self.ip2location_api_version == "keyed":
+                payload["key"] = self._api_key_name
+
+            location_info = self.get_response(payload)
+            location_info.raise_for_status()
+
+        except requests.RequestException as e:
+            raise AnalyzerRunException(e)
+
+        response = {"location_info": location_info.json()}
```

Comment:
no need to create a wrapper dictionary. Just use the response dict itself

Reply:
my bad changed it.